feat(testing): adding a11y testing via chai-a11y-axe plugin (#620)
open-wc/open-wc: packages/chai-a11y-axe/README.md

> # Chai a11y axe

Reviewer: ```suggestion
# Chai A11y aXe
```
either that or 
```md
# `chai-a11y-axe`
```

Reviewer 2: should `A` be capital in `axe`? It doesn't feel right. I'm not native speaker though.

Reviewer: No, since aXe is a sort of brand name. For example, one wouldn't capitalize the m in macOS but they would the w in Windows, because that's how those brands are identified.

Reviewer 2: Sorry, I mean in `a11y`. I agree with `aXe`.

Reviewer: I think it's ok to go either way but personally I'd go with a11y in lowercase, even in a title, as it's more common.

Reviewer 2: Anyway, it is updated as you requested.

[//]: # (AUTO INSERT HEADER PREPUBLISH)

This module provides a Chai plugin to perform automated accessibility tests via axe.

::: tip
This is part of the default [open-wc testing](https://open-wc.org/testing/) recommendation
:::

> ## Test for accessibility

Reviewer: ```suggestion
## Testing for Accessibility
The plugin provides a chai assertion call `accessible` which performs an accessibility test on your element.
It ships with interfaces for both Chai's BDD UI (expect) and TDD UI (assert).
```

> ### Expect library

Reviewer: ```suggestion
### Chai BDD UI
```

> The plugin provides `accessible()` function to perform accessibility test.

Reviewer: ```suggestion
The BDD UI works with chai's `expect` function.
```

> Because the test is asynchronous it has to be performed by returning the value from calling the function, by using async/await, or by using `done()` function that is accepted as an option.

Reviewer: ```suggestion
Because the test is asynchronous, you must either await its result or pass a `done` parameter in the plugin's options object.
```

Rules can be ignored by passing `ignoredRules` with list of ignored rules as a configuration option.

```js
import { fixture, expect, html } from '@open-wc/testing';

it('passes accessibility test', async () => {
  const el = await fixture(html`<button>label</button>`);
  await expect(el).to.be.accessible();
});

it('fails without label', async () => {
  const el = await fixture(html`<div aria-labelledby="test-x"></div>`);
  await expect(el).not.to.be.accessible();
});

it('passes for all rules, ignores attributes test', async () => {
  const el = await fixture(html`<div aria-labelledby="test-x"></div>`);
  await expect(el).to.be.accessible({
    ignoredRules: ['aria-valid-attr-value']
  });
});

it('accepts "done" option', (done) => {
  fixture(html`<button>some light dom</button>`)
  .then((el) => {
    expect(el).to.be.accessible({
      done
    });
  });
});
```

> ### Assert library

Reviewer: ```suggestion
### Chai TDD UI
```

> The plugin provides `isAccessible()` and `isNotAccessible()` function on `assert` library.

Reviewer: ```suggestion
The `isAccessible()` and `isNotAccessible()` methods work on Chai's on `assert` function.
```

```js

> import { fixture, expect, html } from '@open-wc/testing';

Reviewer: ```suggestion
import { fixture, assert, html } from '@open-wc/testing';
```
